pkg/controller/machinepool/machinepool_controller_test.go

```diff
@@ -1428,6 +1487,16 @@ func TestRemoteMachineSetReconcile(t *testing.T) {
 				return
 			}
 
+			// Special validation for NotEnoughReplicas condition clearing test
+			if test.name == "NotEnoughReplicas condition cleared when switching from autoscaling to fixed replicas" {
```

Comment:
Eh, it seems icky to call out a specific test by name here.

I think we can do better, particularly in light of the comment I made on the code side.